include/irr/builtin/material_compiler/glsl/common.glsl

```diff
@@ -0,0 +1,447 @@
+#version 430 core
+
+#include <irr/builtin/glsl/virtual_texturing/extensions.glsl>
+
+layout (location = 0) in vec3 WorldPos;
+layout (location = 1) flat in uint InstanceIndex;
+layout (location = 2) in vec3 Normal;
+layout (location = 3) in vec2 UV;
+
+layout (location = 0) out vec4 OutColor;
+
+#define instr_t uvec2
+#define reg_t vec3
+#define REG_COUNT 16
+#define TEX_PREFETCH_REG_COUNT 8
+#define NORMAL_REG_COUNT 3
+
+//in 16-byte/uvec4 units
+//layout (constant_id = 0) const uint sizeof_bsdf_data = 3;
+#define sizeof_bsdf_data 3
+
+struct bsdf_data_t
+{
+	uvec4 data[sizeof_bsdf_data];
+};
+
+#define _IRR_VT_DESCRIPTOR_SET 0
+#define _IRR_VT_PAGE_TABLE_BINDING 0
+
+#define _IRR_VT_FLOAT_VIEWS_BINDING 1 
+#define _IRR_VT_FLOAT_VIEWS_COUNT 4
+#define _IRR_VT_FLOAT_VIEWS
+
+#define _IRR_VT_INT_VIEWS_BINDING 2
+#define _IRR_VT_INT_VIEWS_COUNT 0
+#define _IRR_VT_INT_VIEWS
+
+#define _IRR_VT_UINT_VIEWS_BINDING 3
+#define _IRR_VT_UINT_VIEWS_COUNT 0
+#define _IRR_VT_UINT_VIEWS
```

Comment:
you want to narrow these down when the BSDF forest doesnt include certain float views

Reply:
done?

Reply:
TODO

Reply:
what happened to this?